src/Microsoft.AspNetCore.DataProtection/KeyManagement/KeyBase.cs

```diff
@@ -41,6 +52,24 @@ public IAuthenticatedEncryptorDescriptor Descriptor
             }
         }
 
+        public IAuthenticatedEncryptor CreateEncryptorInstance()
+        {
+            if (_encryptor == null)
+            {
+                foreach (var factory in _encryptorFactories)
+                {
+                    var encryptor = factory.CreateEncryptorInstance(this);
+                    if (encryptor != null)
+                    {
+                        _encryptor = encryptor;
+                        break;
+                    }
+                }
+            }
+
+            return _encryptor;
```

Comment:
Spoke offline. This lock here was added as a perf optimization and we don't need to make changes to this.